app/queries/latest_metrics.rb

```diff
@@ -0,0 +1,16 @@
+class LatestMetrics < SimpleDelegator
+  def find
+    Metric.find_by_sql([sql, id])
+  end
+
+  def sql
+    'SELECT m1.*
+    FROM metrics m1
+    WHERE m1.branch_id = ?
```

Comment:
The thing is that I need the first metric for each group and I can not do that using a group by. Can you find a way to do it?

I want the latest metric grouped by name